README.md

```diff
@@ -176,6 +177,92 @@ try {
 }
 ```
 
+#### Type guards
+
+There are several [type guards](https://www.typescriptlang.org/docs/handbook/advanced-types.html#type-guards-and-differentiating-types)
```

Comment:
this link is deprecated according to Typescript. should we supply the most up to date link instead?

https://www.typescriptlang.org/docs/handbook/2/narrowing.html